Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+In the rare case that you need to make a bundle from a package you're developing locally, use `devtools::build()`. Under the hood, this calls `pkgbuild::build()` and, ultimately, `R CMD build`, which is described further in the [Building package tarballs][wre-tarball] secion of [Writing R Extensions][wre].
+
+Every CRAN package is available in this form, via the "Package source" field of individual landing pages. Continuing our examples from above, you could download the bundles `forcats_0.4.0.tar.gz` and `stringr_1.4.0.tar.gz` (or whatever the current versions may be). You could unpack such a bundle in the shell (not the R console) like so:
+
+```bash
+tar xvf forcats_0.4.0.tar.gz
+```
+
+If you decompress a bundle, you'll see it looks almost the same as a source package. This diagram summarises the files present in the top-level directory for source, bundled, and binary versions of devtools.
+
+*TODO: Remake this, possibly with a different (fictional or deliberately crafted?) package. Things I want to address: README.Rmd, .github/, CI config, pkgdown, LICENSE.md, data-raw/, updated treatment of NEWS.md, existence of vignettes/ in tarball, presence of build/ in tarball, absence of README.md in binary form.*
+
+```{r, echo = FALSE}
+knitr::include_graphics("diagrams/package-files.png")
+```
+
+The main differences between a source package and an uncompressed bundle are:
+
+* Vignettes have been built, so rendered outputs, such as HTML, appear below
+  `inst/doc/` and a vignette index appears in the `build/` directory, usually
+  alongside a PDF package manual.
+
+* A local source package might contain temporary files used to save time during
+  development, like compilation artefacts in `src/`. These are never found in 
+  a bundle.
+
+* Any files listed in `.Rbuildignore` are not included in the bundle. These are
+  typically files that facilitate your development process, but that should be
+  excluded from the distributed product.
+
+#### `.Rbuildignore` {#rbuildignore}
+
+You won't need to contemplate the exact structure of package `.tar.gz` files very often, but you do need to understand the `.Rbuildignore` file. It controls which files from the source package make it into the downstream forms.
+
+Each line of `.Rbuildignore` is a Perl-compatible regular expression that is matched, without regard to case, against the path to each file in the source package, i.e. the result of `dir(full.names = TRUE, recursive = TRUE, include.dirs = TRUE, all.files = TRUE)` run from the package's top-level directory. If the regular expression matches, that file or directory is excluded. Note there are some default exclusions implemented by R itself, mostly relating to classic version control systems and editors, such as SVN, Git, and Emacs.
```

Comment:
The 1st edition alluded to `dir(full.names = TRUE)` re: listing all the relevant filepaths. But if we're going to give literal R code, I think what I give above is more accurate. Which makes me question whether we should really give code or just use prose.

Reply:
Maybe include the code in a footnote? It's not that important that people know it, but it might be useful for debugging a wonky `.Rbuildignore`

Reply:
Footnote-ized.